tests/conftest.py

```diff
@@ -120,8 +120,14 @@ def pytest_collection_modifyitems(items):
         http://pytest.org/latest/plugins.html
     """
     for item in items:
-        if 'qtbot' in getattr(item, 'fixturenames', ()):
+        if 'qapp' in getattr(item, 'fixturenames', ()):
             item.add_marker('gui')
+            if sys.platform == 'linux' and not os.environ.get('DISPLAY', ''):
+                if 'CI' in os.environ:
+                    raise Exception("No display available on CI!")
```

Comment:
I understand you don't even want to run the tests if no display is available, but as a side note, if you want to make only tests which require `qapp` to fail, you can move this code to a `pytest_runtest_setup` hook.